src/Chapter2.hs

```diff
@@ -620,7 +653,19 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [Int] -> [Int]
+takeEven [] = []
+takeEven [x] = [x]
+takeEven l = checkEven i l
+    where
+      i = 0
+      checkEven :: Int -> [Int] -> [Int]
+      checkEven i l
+        | l == [] = []
+        | (mod i 2 == 0) && i < length l = (l !! i) : checkEven (i+1) l
+        | (mod i 2 /= 0) && i < length l = checkEven (i+1) l
+        | otherwise = []
```

Comment:
Creative solution!
Though, Here you can use the pattern matching (similarly to what you did in the duplicate, but you will need one more case to examine) for a more faster solution. 🚤